rfcs/0050-vm-syscalls-3/0050-vm-syscalls-3.md

```diff
@@ -12,174 +12,139 @@ Created: 2023-04-17
+This syscall retrieves the file descriptors available to the current process, which are passed in from the parent process. These results are copied from the `inherited_fds` parameter of the Spawn syscall.
 
 ```c
-int ckb_get_memory_limit();
+int ckb_inherited_file_descriptors(uint64_t* fd, size_t* count);
 ```
 
-For the script(prime script) directly invoked by CKB, it will always return 8(4M). For the child script invoked by prime script or other child script, it depends on the parameters set by *Spawn*.
+When returning from the syscall, the syscall fills `fd` with the file descriptors in unit of `uint64_t` and fills in `count` with the count of corresponding file descriptors. The actual count of file descriptor written to `fd` is the minimum value between the count of `inherited_fds` in the Spawn syscall and the input value pointed to by `count` before syscall.
+
+By providing `NULL` as `fd` and a `size_t` pointer with a value of 0 as `count`, this syscall can be used to fetch the count of `inherited_fds` in the Spawn syscall. However, by providing `NULL` as `fd` and a non-zero value as `count`, this syscall fails.
 
-### Current Memory
-[Current Memory]: #current-memory
+### Read
+[Read]: #read
 
-Get the Current Memory Usage. The result is the sum of the memory usage of the parent script and the child script.
+This syscall reads data from a pipe via a file descriptor. The syscall Read attempts to read up to value pointed by length bytes from file descriptor fd into the buffer, and the actual length of data read is written back to the length parameter. The syscall may pause the execution of current process.
 
 ```c
-int ckb_current_memory();
+int ckb_read(uint64_t fd, void* buffer, size_t* length);
 ```
 
-The system call returns an integer. Multiply it by 0.5 to get the actual memory usage (unit: megabytes)
+### Write
+[Write]: #write
 
-### Set Content
-[Set Content]: #set-content
-
-The child script can return bytes data to the parent script through `Set Content`.
+This syscall writes data to a pipe via a file descriptor. The syscall Write writes up to value pointed by length bytes from the buffer, and the actual length of data written is written back to the length parameter. The syscall may pause the execution of current process.
 
 ```c
-int ckb_set_content(uint8_t* content, uint64_t* length);
+int ckb_write(uint64_t fd, const void* buffer, size_t* length);
 ```
 
-- Length up to 256K.
-- If the written length is greater than the limit given by *Spawn*, the final written length is the minimum of the two.
-- Returns 0 for success, otherwise for failure.
-- This function is optional. Not every child script needs to call this, and nothing will happen if you call it from the prime script.
+### Close
+[Close]: #close
 
-### Spawn example
+This syscall manually closes a file descriptor. After calling this, any attempt to read/write the file descriptor pointed to the other end would fail.
```

Comment:
Like mentioned above, please describe in details how read/write to a closed channel would fail. This can be a very useful utility.